### PR TITLE
Homework11: Ekaterina Zavadskaya

### PR DESCRIPTION
**Pull request by KateZavadskaya**
Add done homework11

Docs: hm11_1.py, hm11_2.py

### CODE REVIEW

File: hm11_1.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+""" Task 2: bank deposit """
+
+
+class Book:
+
+    """ h """
```
Comment (Stml89) [inline hm11_1.py]: Please write good docstring

File: hm11_1.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    rsrved = False
+    taken = False
+
+    def about_book(self):
+
+        """ h """
+
+        return f"{self.book_name} Врезерве:{self.rsrved}, наруках:{self.taken}"
+
+    # def take_book_status(self):
```
Comment (Stml89) [inline hm11_1.py]: remove ti

File: hm11_1.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    def x2(self):
+
+        """  for pylint :(, no need in code """
+
+
+book_user_1 = BookUser("Катя")
+book_user_2 = BookUser("Дима")
+
+
+book_1 = Book()
```
Comment (Stml89) [inline hm11_1.py]: pass book information here, and do not hardcode it in the class

File: hm11_1.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    #     if self.rsrved == False:
+    #         return f"Книгу можно зарезервировать"
+    #     else:
+    #         return f"Книгу нельзя зарезервировать"
+
+    def take_book(self):
+
+        """ h """
+
+        if self.taken is False:
```
Comment (Stml89) [inline hm11_1.py]: `if not self.taken:`

File: hm11_2.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    """ h """
+    persent = 30
+
+    def __init__(self, summ_dep, dep_time, dep_n):
+        self.summ_dep = summ_dep
+        self.dep_time = dep_time * 12
+        self.dep_n = dep_n
+
+    def dep(self, persent):
```
Comment (Stml89) [inline hm11_2.py]: typo `persent`

File: hm11_2.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
+""" Task 2: bank deposit """
+
+
+class Bank:
+
+    """ h """
+    persent = 30
```
Comment (Stml89) [inline hm11_2.py]: where are you using this class attribute?

File: hm11_2.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        self.dep_time = dep_time * 12
+        self.dep_n = dep_n
+
+    def dep(self, persent):
+
+        """ j """
+        self.persent = persent
+        monthly_persent = persent/100/12
+        persent_total = self.summ_dep * (1 + monthly_persent) ** self.dep_time
+        return persent_total
```
Comment (Stml89) [inline hm11_2.py]: round final amount to 2 digits after `.`

File: hm11_1.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+        """ for pylint :(, no need in code """
+
+    def x2(self):
+
+        """  for pylint :(, no need in code """
+
+
+book_user_1 = BookUser("Катя")
+book_user_2 = BookUser("Дима")
```
Comment (Stml89) [inline hm11_1.py:80]: how are you using this 2 users?

In general, user come to the library and trying to do something with books:
``
book_user_1.reserve_book(book1)
book_user_2.reserve_book(book2)

book_user_1.take_book(book1)
book_user_2.take_book(book2)
``

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hm11_1.py (hunk 1 above)
Comment (KateZavadskaya) [reply hm11_1.py]: Done

File: hm11_1.py (hunk 2 above)
Comment (KateZavadskaya) [reply hm11_1.py]: Made a minor fix, but leave it in the code - think it should be a part of general logit and code.

File: hm11_1.py (hunk 3 above)
Comment (KateZavadskaya) [reply hm11_1.py]: But it hardcode tatol result, as for me. I think it is easier once write it in code, then write in result.
I understand your point!) Will also do my best about this point in next HMs.

File: hm11_1.py (hunk 4 above)
Comment (KateZavadskaya) [reply hm11_1.py]: Done

File: hm11_2.py (hunk 5 above)
Comment (KateZavadskaya) [reply hm11_2.py]: Done

File: hm11_2.py (hunk 6 above)
Comment (KateZavadskaya) [reply hm11_2.py]: It is the Pylinter wishes :(
hm11_2.py:17:8: W0201: Attribute 'percent' defined outside init (attribute-defined-outside-init)

File: hm11_2.py (hunk 7 above)
Comment (KateZavadskaya) [reply hm11_2.py]: Done

File: hm11_1.py (hunk 8 above)
Comment (KateZavadskaya) [reply hm11_1.py:80]: Make some changes, now it seems to library user.

File: hm11_1.py (hunk 3 above)
Comment (Stml89) [reply hm11_1.py]: Please, fix it here(this HW)

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hm11_1.py (hunk 3 above)
Comment (KateZavadskaya) [reply hm11_1.py]: Just tried to explain my point of view. And may be was not correct in my message 🙏
Got you!) Will fix! 👍

File: hm11_2.py (hunk 6 above)
Comment (Stml89) [reply hm11_2.py]: why you can not move it to `__init__`?
```
    def __init__(self, summ_dep, dep_time, dep_n, persent=0.1):
        self.summ_dep = summ_dep
        self.dep_time = dep_time * 12
        self.dep_n = dep_n
        self.persent = persent
```

File: hm11_1.py (hunk 2 above)
Comment (Stml89) [reply hm11_1.py]: The goal is to have a clear codebase and keep only what you are using, so use it or remove

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hm11_1.py (hunk 2 above)
Comment (KateZavadskaya) [reply hm11_1.py]: @Stml89 , add persent to __init__(11_2), also delete extra code (11_1).
Pylint / build (pull_request) is faild, yes - "R0913: Too many arguments (8/5) (too-many-arguments)". Have already disscissed with you this in Discord, please. 
Plaese, check my push despite Pylint🙏

**Review by Stml89 — APPROVED** [review]

**Pull request merged**